modules/representation_management/spec/controllers/power_of_attorney_controller_spec.rb

```diff
@@ -0,0 +1,114 @@
+    end
+
+    describe 'GET #index' do
+      it 'returns a successful response' do
+        allow(service).to receive(:get_power_of_attorney).and_return({ 'data' => { 'type' => 'organization',
+                                                                                   'attributes' => { 'code' => 'og1' } } }) # rubocop:disable Layout/LineLength
+        allow(controller).to receive(:find_poa_by_code).and_return(organization)
+        get :index
+
+        expect(response).to be_successful
+      end
+
+      it 'returns the expected response when a user has a POA' do
+        expected_serialized_organization = ActiveModelSerializers::SerializableResource.new(
+          organization,
+          serializer: RepresentationManagement::PowerOfAttorney::OrganizationSerializer
+        ).to_json
+        allow(service).to receive(:get_power_of_attorney).and_return({ 'data' => { 'type' => 'organization',
+                                                                                   'attributes' => { 'code' => 'og1' } } }) # rubocop:disable Layout/LineLength
+        allow(controller).to receive(:find_poa_by_code).and_return(expected_serialized_organization)
+        get :index
+
+        expect(response.body).to eq(expected_serialized_organization)
+      end
+
+      it 'returns the expected response when a user does not has a POA' do
+        allow(service).to receive(:get_power_of_attorney).and_return({})
+        get :index
+
+        expect(response.body).to eq("{}") # rubocop:disable Style/StringLiterals
+      end
+    end
+
+    describe 'private methods' do
+      it 'finds organization poa by code' do
+        allow(Veteran::Service::Organization).to receive(:find).with('og1').and_return(organization)
+        expected_serialized_organization = ActiveModelSerializers::SerializableResource.new(
+          organization,
+          serializer: RepresentationManagement::PowerOfAttorney::OrganizationSerializer
+        ).as_json
+
+        expect(controller.send(:find_poa_by_code, 'organization', 'og1')).to eq(expected_serialized_organization)
+      end
+
+      it 'finds individual and representative poa by code' do
+        allow(Veteran::Service::Representative).to receive(:find).with('rp1').and_return(representative)
+        expected_serialized_representative = ActiveModelSerializers::SerializableResource.new(
+          representative,
+          serializer: RepresentationManagement::PowerOfAttorney::RepresentativeSerializer
+        ).as_json
+
+        %w[individual representative].each do |type|
+          expect(controller.send(:find_poa_by_code, type, 'rp1')).to eq(expected_serialized_representative)
+        end
+      end
+
+      it 'finds organization' do
+        allow(Veteran::Service::Organization).to receive(:find).with('og1').and_return(organization)
+        expect(controller.send(:find_organization, 'og1')).to eq(organization)
+      end
+
+      it 'finds representative' do
```

Comment:
The outside network calls and services (calling LH, identity providers, etc) are most likely the only interactions we want to stub/mock in these unit tests.

The rep and org records are created in the db and if we want to test finding them, then we want to let the AR query run naturally. Overriding the return doesn't actually test the search.

Reply:
This test has been removed.